### PR TITLE
Optimization: Lazy model generation 

### PR DESCRIPTION
**Pull request by elsh**
Moved model creation to when actually needed

### CODE REVIEW

File: Sources/MockoloFramework/Models/ClassModel.swift [hunk 1]
```diff
@@ -46,7 +46,8 @@ final class ClassModel: Model {
         self.needInit = needInit
         self.initParams = initParams
         self.offset = offset
-        self.attribute = Set(attributes).joined(separator: " ")
+        let ret = attributes.filter {$0.contains(String.available)}
```
Comment (benpious) [inline Sources/MockoloFramework/Models/ClassModel.swift]: This has been bothering me for a while; what is `ret`? "Return?" Can we use a more descriptive name like `availableAttributes`?

File: Sources/MockoloFramework/Models/MethodModel.swift [hunk 2]
```diff
@@ -134,8 +134,9 @@ final class MethodModel: Model {
         assert(paramDecls.count == comps.count)
         
         let zippedParams = zip(paramDecls, comps)
-        self.params = zippedParams.map { (argAst: Structure, argLabel: String) -> ParamModel in
-            ParamModel(argAst, label: argLabel, offset: argAst.offset, length: argAst.length, data: data, isInitializer: ast.isInitializer)
+        self.params = zippedParams.map { (arg: (Structure, String)) -> ParamModel in
```
Comment (benpious) [inline Sources/MockoloFramework/Models/MethodModel.swift]: You probably don't need this. Swift generally only gives this diagnostic when it needs it to give the real error.

File: Sources/MockoloFramework/Models/ParamModel.swift [hunk 3]
```diff
@@ -24,39 +24,39 @@ final class ParamModel: Model {
     var type: Type
     let label: String
     let isGeneric: Bool
-    let isInitializer: Bool
+    let inInit: Bool
```
Comment (benpious) [inline Sources/MockoloFramework/Models/ParamModel.swift]: What's the purpose of this change?

Comment (elsh) [reply Sources/MockoloFramework/Models/ParamModel.swift]: this description is more accurate than is*

File: Sources/MockoloFramework/Models/ClassModel.swift (hunk 1 above)
Comment (elsh) [reply Sources/MockoloFramework/Models/ClassModel.swift]: yes
